### PR TITLE
Add olafurpg setup-java, sbt-ci-release & gpg-setup

### PR DESCRIPTION
**Pull request by gvolpe**
We are now using this to make releases at the company I work for (private repo). This works but I'm open to suggestions on how to improve it :)

Here's a link to the plugin: https://github.com/olafurpg/sbt-ci-release

### CODE REVIEW

**Comment by gvolpe** [comment]
I just realized that [setup-scala](https://github.com/olafurpg/setup-scala) is now listed as [setup-java](https://github.com/marketplace/actions/setup-java) in the marketplace, so it is redundant. I'll fix this soon.

**Comment by gvolpe** [comment]
It is now being successfully used here https://github.com/profunktor/redis4cats :)

File: steps/java-setup.dhall [hunk 1]
```diff
@@ -4,13 +4,11 @@ in    λ(args : { java-version : Text })
     → Step::{
       , id = None Text
       , name = Some "java ${args.java-version} setup"
-      , uses = Some "actions/setup-java@v1"
+      , uses = Some "olafurpg/setup-java@v6"
```
[the remark this answers is not in the document]
Comment (gvolpe) [reply steps/java-setup.dhall:7]: Sounds good to me :+1: . It gives better visibility of what actions package we are actually using. Do you want me to give it a try?

I can rename this new one to `steps.olafurpg/setup-java` as suggested and modify the others to have as `steps.actions/setup-java` for example? Or you would prefer to leave them as they are now in `master`?

[the remark this answers is not in the document]
Comment (regadas) [reply steps/java-setup.dhall:7]: @gvolpe I can go ahead and do it, should be fairly quick. I'll probably keep them on master for now.

**Comment by regadas** [comment]
@gvolpe Thanks!

**Comment by gvolpe** [comment]
Thank you! :tada:

**Comment by gvolpe** [comment]
I've been thinking that the `sbt-ci-release` shouldn't take arguments since they should always be the same so we can have them defined in the implementation straight away:

```dhall
let Step = ../schemas/Step.dhall

in    Step::{
      , id = None Text
      , name = Some "Publish \${{ github.ref }}"
      , uses = None Text
      , run = Some "sbt ci-release"
      , env = Some
          ( toMap
              { PGP_PASSPHRASE = "\${{ secrets.PGP_PASSPHRASE }}"
              , PGP_SECRET = "\${{ secrets.PGP_SECRET }}"
              , SONATYPE_PASSWORD = "\${{ secrets.SONATYPE_PASSWORD }}"
              , SONATYPE_USERNAME = "\${{ secrets.SONATYPE_USERNAME }}"
              }
          )
      }
```

WDYT?

**Pull request merged**